lib/reek/configuration/configuration_validator.rb

```diff
@@ -0,0 +1,35 @@
+module Reek
+  module Configuration
+    #
+    # Configuration validator module.
+    #
+    module ConfigurationValidator
+      private
+
+      def smell_type?(key)
+        Reek::Smells.const_get key
+      rescue NameError
+        false
+      end
+
+      def error_message_for_missing_directory(pathname)
+        "Configuration error: Directory `#{pathname}` does not exist"
+      end
+
+      def error_message_for_file_given(pathname)
+        "Configuration error: `#{pathname}` is supposed to be a directory but is a file"
+      end
+
+      def validate_directory(pathname)
+        abort(error_message_for_missing_directory(pathname)) unless pathname.exist?
+        abort(error_message_for_file_given(pathname)) if pathname.file?
+      end
```

Comment:
Because a "raise" would give an ugly, useless stacktrace. The "abort" happens on a configuration error done by the user and we should make it as easy as possible for him to see what's going on.
A stacktrace is far more deterring than just a proper error message.
P.S.: 
I believe we should use "abort" instead of raise for all errors that are presented to the user but that would be another PR / discussion.